### PR TITLE
Add network peer address attributes to the span for the `okhttp-3.0` instrumentation

### PR DESCRIPTION
**Pull request by serkan-ozal**

### CODE REVIEW

**Comment by serkan-ozal** [comment]
After I rebased from main, `Vaadin14LatestTest` started failing, so not sure it is related to my changes.

**Comment by trask** [comment]
> After I rebased from main, `Vaadin14LatestTest` started failing (was passing before just yesterday), so not sure it is related to my changes.

hey @serkan-ozal, this should be fixed in main just now (#12025), if you rebase once more it should get cleared

**Comment by serkan-ozal** [comment]
Thanks @trask, it is green now. May I also get your review too :)

File: instrumentation/okhttp/okhttp-3.0/testing/src/main/java/io/opentelemetry/instrumentation/okhttp/v3_0/AbstractOkHttp3Test.java [hunk 1]
```diff
@@ -180,7 +181,16 @@ public void onResponse(Call call, Response response) {
         trace -> {
           trace.hasSpansSatisfyingExactly(
               span -> span.hasName("parent").hasKind(SpanKind.INTERNAL).hasNoParent(),
-              span -> span.hasName("GET").hasKind(SpanKind.CLIENT).hasParent(trace.getSpan(0)),
```
Comment (laurit) [inline instrumentation/okhttp/okhttp-3.0/testing/src/main/java/io/opentelemetry/instrumentation/okhttp/v3_0/AbstractOkHttp3Test.java:183]: I think we can leave this as it was. The network attributes are already tested in AbstractHttpClientTest.

Comment (serkan-ozal) [reply instrumentation/okhttp/okhttp-3.0/testing/src/main/java/io/opentelemetry/instrumentation/okhttp/v3_0/AbstractOkHttp3Test.java]: @laurit done

**Review by laurit — APPROVED** [review]

**Pull request merged**